### PR TITLE
rewrite the text classification demo.

### PR DESCRIPTION
**Pull request by lcy-seso**
rewrite the text classification example.

### CODE REVIEW

File: image_classification/train.py [hunk 1]
```diff
@@ -51,10 +37,10 @@ def main():
         learning_rate_schedule="discexp", )
 
     train_reader = paddle.batch(
-        paddle.reader.shuffle(reader.test_reader("train.list"), buf_size=1000),
+        paddle.reader.shuffle(reader.train_reader("train.list"), buf_size=1000),
```
Comment (llxxxll) [inline image_classification/train.py:40]: train.list是文本数据集吗？没有在目录下找到

File: image_classification/train.py [hunk 2]
```diff
@@ -51,10 +37,10 @@ def main():
...
         batch_size=BATCH_SIZE)
     test_reader = paddle.batch(
-        reader.train_reader("test.list"), batch_size=BATCH_SIZE)
+        reader.test_reader("test.list"), batch_size=BATCH_SIZE)
```
Comment (llxxxll) [inline image_classification/train.py:43]: test.list是文本数据集吗？没有在目录下找到

File: text_classification/index.html [hunk 3]
```diff
@@ -41,243 +41,196 @@
...
+│   └── dnn_net.png
+├── index.html
+├── infer.py            # 预测任务脚本
+├── network_conf.py     # 本例中涉及的各种网络结构均定义在此文件中，希望进一步修改模型结构，请修改此文件
+├── reader.py           # 读取数据接口，若使用自定义格式的数据，可直接修改此文件
+├── README.md           # 文档
+├── run.sh              # 运行此脚本，可以以默认参数直接开始训练任务
+├── train.py            # 训练任务脚本
+└── utils.py            # 定义通用的函数，例如：打印日志、解析命令行参数、构建字典、加载字典等
+```
```
Comment (llxxxll) [inline text_classification/index.html]: 缺少一个快速开始。其中包含上面的目录说明，及一个训练说明（训练过程输出日志里都是什么意思）、和一个预测说明（可以是一段code，参考：https://www.oschina.net/p/jieba/?fromerr=btIKdxHH -功能 1)分词的代码及output、当然也可以是一个gif图）

File: text_classification/index.html [hunk 4]
```diff
@@ -41,243 +41,196 @@
...
 
-    # fc and output layer
-    output = paddle.layer.fc(
-        input=[conv_3, conv_4], size=class_dim, act=paddle.activation.Softmax())
+例如有如下格式的数据：每一行为一条样本，以 `\t` 分隔，第一列是类别标签，第二列是输入文本的内容。文本内容中的词语以空格分隔。以下是两条示例数据：
 
-    cost = paddle.layer.classification_cost(input=output, label=lbl)
-
-    return cost, output, lbl
+```
```
Comment (llxxxll) [inline text_classification/index.html]: 如果是面向初学者，数据格式的自定义要比其他信息重要得多，因为这可能是唯一"需要考虑"的事，所以优先级需要提到快速开始后面。而详解的这个位置可以对应加一个锚点链接。

File: text_classification/index.html (hunk 3 above)
Comment (llxxxll) [reply text_classification/index.html]: 接下来应该是之前写在末尾的『修改参数说明』，这样可以在新手运行出一个结果的前提下，来对比修改不同参数得到的结果不同。于是顺势引出CNN&DNN该如何选择

Comment (llxxxll) [reply text_classification/index.html]: 其实余下部分都是在解释这些『参数』的含义，顺序也应如此。

File: text_classification/run.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+python train.py \
+--nn_type="dnn" \
```
Comment (llxxxll) [inline text_classification/run.sh:4]: 这里的train的方式怎么又改成shell传参了，按照约定都应该写到train.py里？

File: text_classification/train.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+import paddle.v2 as paddle
+
+import network_conf
+import reader
+from utils import *
+
+
+def train(topology,
+          train_data_dir=None,
```
Comment (llxxxll) [inline text_classification/train.py:14]: lm_rnn.py 里有个`run_type=GRU #'or LSTM'`，在这里是否可以增加一个方法的选择参数比如`DNN OR CNN`

File: image_classification/train.py (hunk 2 above)
Comment (lcy-seso) [reply image_classification/train.py:43]: 这处修改用来修改图像分类例子的bug，目前每个例子读取数据的方式确实不统一。后续提PR修改图像分类的例子。

File: image_classification/train.py (hunk 1 above)
Comment (lcy-seso) [reply image_classification/train.py:40]: 这处修改用来修改图像分类例子的bug，目前每个例子读取数据的方式确实不统一。后续提PR修改图像分类的例子。

**Review by lcy-seso — COMMENTED** [review]
follow comments.

File: text_classification/run.sh (hunk 5 above)
Comment (lcy-seso) [reply text_classification/run.sh:4]: shell 里面为 `train.py` 指定的参数，直接运行shell 即可，否则需要在命令行敲长串的参数。

File: text_classification/train.py (hunk 6 above)
Comment (lcy-seso) [reply text_classification/train.py:14]: 已有此参数，`nn_type`  用来指定选择何种模型。

**Comment by luotao1** [comment]
Readme的目录顺序需要调整么？把模型介绍、模型详解放到最开头的地方？

**Review by llxxxll — APPROVED** [review]
